File: 2.Base-Check/sol2.js
function isValidNumber(n, base) {
  if (base < 2 || base > 36) return false;

  const digits = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
  const allowedSet = new Set(digits.slice(0, base));

  const upperN = n.toUpperCase();

  for (const char of upperN) {
    if (!allowedSet.has(char)) return false;
  }

  return true;
}

File: 2.Base-Check/sol2.py
def is_valid_number(n, base):
    if base < 2 or base > 36:
        return False

    digits = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    allowedSet = set(digits[:base])

    for char in n.upper():
        if char not in allowedSet:
            return False
        
    return True